agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -33,7 +33,7 @@ def initialize(autostart = true)
     end
 
     def start
-      observe(Actor[:node_info_worker]) do |node|
+      observe(Actor[:node_info_worker].observable) do |node|
         launch(node)
```

Comment:
Breaking change with the `exclusive` yield:

```
E, [2017-08-29T15:45:59.682615 #1] ERROR -- Kontena::NetworkAdapters::Weave: RuntimeError: Cannot execute blocks on sender in exclusive mode
E, [2017-08-29T15:45:59.682726 #1] ERROR -- Kontena::NetworkAdapters::Weave: /usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:13:in `initialize'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/call/sync.rb:8:in `initialize'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/proxy/sync.rb:20:in `new'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/proxy/sync.rb:20:in `method_missing'
/app/lib/kontena/network_adapters/weave.rb:172:in `ensure_exposed'
/app/lib/kontena/network_adapters/weave.rb:268:in `post_start'
/app/lib/kontena/network_adapters/weave.rb:227:in `launch'
/app/lib/kontena/network_adapters/weave.rb:37:in `block in start'
/app/lib/kontena/observer.rb:290:in `block in observe'
/app/lib/kontena/observer.rb:225:in `block in receive'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:97:in `exclusive'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid.rb:421:in `exclusive'
/app/lib/kontena/observer.rb:224:in `receive'
/app/lib/kontena/observer.rb:280:in `observe'
/app/lib/kontena/observer.rb:92:in `observe'
/app/lib/kontena/observer.rb:29:in `observe'
/app/lib/kontena/network_adapters/weave.rb:36:in `start'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `public_send'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/call/async.rb:7:in `dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:50:in `block in dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:76:in `block in task'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'
```

OTOH I think the answer is to fix the weave actor `launch` method to work in exclusive mode by working around the celluloid limitation on call blocks in exclusive mode.